Test: Fixture for path to example GEFF file

melisande-c opened the pull request:
# Proposed Change
Add pytest fixture for a path to example data and the expected graph attributes. This can be used as an initial first pass test for new backends, an example test for the networkx API has been added.

# Types of Changes
- Tests

# Checklist
Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.

- [x] I have read the developer/contributing docs.
- [x] I have added tests that prove that my feature works in various situations or tests the bugfix (if appropriate).
- [x] I have checked that I maintained or improved code coverage.
- [ ] I have written docstrings and checked that they render correctly.
- [ ] If I changed the specification, I have checked that any validation functions and tests reflect the changes.

msschwartz21 commented on tests/test_networkx/conftest.py:
```diff
@@ -0,0 +1,123 @@
+
+
+@pytest.fixture()
+def path_w_expected_graph_attrs(tmp_path) -> tuple[Path, GraphAttrs]:
```
I'm wondering if it would be better to just have this as a normal function so that we can take advantage of the parametrization of different dtypes?

msschwartz21 commented on tests/test_networkx/conftest.py:
```diff
@@ -0,0 +1,123 @@
+        directed=directed,
+    )
+
+    # TODO: write zarr from scratch to remove test dependency on networkx backend?
```
I'm ok with doing this through networkx because I suspect we will always ship geff minimally with networkx

msschwartz21 reviewed (changes requested):
This looks great! Let's move conftest.py into the tests/ root so that it's available to the other backends. Otherwise I think we should fully refactor the networkx tests to use this new framework 🎉

melisande-c commented:
> Let's move conftest.py into the tests/ root so that it's available to the other backends. 

Oops! I have fixed.

I have added the requested changes. The fixture has been updated to be a function to make it parameterisable, it is still a fixture so that it can automatically receive the `tmp_path` fixture. I refactored the original networkx tests to use this new fixture, and removed the backend template function. I will add an issue to add documentation for how new backends can use this for their own tests!
